### PR TITLE
Feat/switch to cached client

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
Pull request covers logic to migrate from CashCtrlClient class to CachedCashCtrlClient class.
This changes increase performance of the package

Attention!: Test will give us en error until https://github.com/macxred/cashctrl_api/pull/28 PR is merged

@lasuk Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

File: cashctrl_ledger/ledger.py [hunk 1]
```diff
@@ -181,35 +180,15 @@ def add_account(self, account: str, currency: str, text: str, group: str, vat_co
...
-        category_id = categories_map[group]
-
         payload = {
             "number": account,
-            "currencyId": currency_id,
+            "currencyId": self._client.currency_to_id(currency),
             "name": text,
-            "taxId": tax_id,
-            "categoryId": category_id,
+            "taxId": self._client.tax_code_to_id(vat_code, allow_missing=True),
```
Comment (lasuk) [inline cashctrl_ledger/ledger.py:187]: I believe `vat_code=None` is not covered. 
If `vat_code is not None`, the tax code has to exist.

I suggest
```python
    "taxId": None if vat_code is None else self._client.tax_code_to_id(vat_code),
```

Instead of:
```python
    "taxId": self._client.tax_code_to_id(vat_code, allow_missing=True),
```

Comment (lasuk) [reply cashctrl_ledger/ledger.py:187]: Same for `update_account()`.

File: cashctrl_ledger/ledger.py [hunk 2]
```diff
@@ -375,17 +318,15 @@ def ledger(self) -> pd.DataFrame:
...
         individual = ledger[ledger['type'] != 'COLLECTIVE']
         result = pd.DataFrame({
             'id': individual['id'],
             'date': individual['dateAdded'].dt.date,
-            'account': [account_map[account] for account in individual['creditId']],
-            'counter_account': [account_map[account] for account in individual['debitId']],
+            'account': [self._client.account_from_id(id) for id in individual['creditId']],
+            'counter_account': [self._client.account_from_id(id) for id in individual['debitId']],
             'amount': individual['amount'],
             'currency': individual['currencyCode'],
```
Comment (lasuk) [inline cashctrl_ledger/ledger.py:331]: Does 'currencyCode' contain the currency in string format ('USD', 'EUR', etc.)?

Is there also a field 'currency' with the currency is text?

File: tests/test_account_chart.py [hunk 3]
```diff
@@ -217,7 +216,7 @@ def test_mirror_accounts(add_and_delete_vat_code):
         "currency": ["CHF"],
         "text": ["2test_account_api_added"],
         "vat_code": ["TestCodeAccounts"],
-        "group": ["/Assets/Anlagevermögen/xyz"],
```
Comment (lasuk) [inline tests/test_account_chart.py:220]: Why do you drop category need '/xyz'?

File: tests/test_account_chart.py [hunk 4]
```diff
@@ -6,7 +6,6 @@
 import requests
```
Comment (lasuk) [inline tests/test_account_chart.py:6]: We're missing e test that ensures adding/updating an account with `vat_code=None` works (and assigns no no VAT code).

**Review by lasuk — COMMENTED** [review]
Thanks. It is a pleasure to see how these changes make the code much more readable while improving performance at the same time.

I've added commit da2df0d to fix broken tests and left a few comments. Back to you, @AlexTheWizardL

File: cashctrl_ledger/ledger.py (hunk 2 above)
Comment (AlexTheWizardL) [reply cashctrl_ledger/ledger.py:331]: Yeah, I`ve checked this out, no need to map this field

File: tests/test_account_chart.py (hunk 4 above)
Comment (AlexTheWizardL) [reply tests/test_account_chart.py:6]: I guess this part is covered in `test_account_mutators` test suit

1. We're adding the account with `None` Vat code
```python
    new_account = {
        'account': 1146,
        'currency': 'CHF',
        'text': 'test create account',
        'vat_code': None,
        'group': '/Assets/Anlagevermögen'
    }
    cashctrl_ledger.add_account(**new_account)
```
2.  Updating that Vat code with existent one
```python
    new_account = {
        'account': 1146,
        'currency': 'CHF',
        'text': 'test update account',
        'vat_code': 'TestCodeAccounts',
        'group': '/Assets/Anlagevermögen'
    }
    cashctrl_ledger.update_account(**new_account)
```


And the same we do with update part:
```python
    new_account = {
        'account': 1145,
        'currency': 'USD',
        'text': 'test update account without VAT',
        'vat_code': None,
        'group': '/Assets/Anlagevermögen'
    }
    cashctrl_ledger.update_account(**new_account)
```

Isn`t this not enough for that?

Comment (lasuk) [reply tests/test_account_chart.py:6]: Thanks. I had somehow missed this.
